deepchecks/tabular/dataset.py

```diff
@@ -1061,6 +1065,172 @@ def datasets_share_date(cls, *datasets: 'Dataset') -> bool:
+            columns = [*columns, label_name]
+
+        is_index_name_str = isinstance(index_name, str)
+        is_index_name_int = isinstance(index_name, int)
+        is_multiindex = isinstance(data.index, pd.MultiIndex)
+
+        unamed_index_levels = {
+            level
+            for level, name in enumerate(data.index.names)
+            if name is None
+        }
+
+        if index_name in data.columns:
+            columns = [index_name, *columns]
+
+        elif (is_index_name_str or is_index_name_int) and index_name in data.index.names:
+            data = data.reset_index()
+            columns = [index_name, *columns]
+
+        elif is_index_name_int and is_multiindex and index_name in unamed_index_levels:
+            data = data.reset_index()
+            index_name = f'level_{index_name}'
+            columns = [index_name, *columns]
+
+        else:
+            data = data.reset_index()
+            index_name = 'index'
+            columns = [index_name, *columns]
+
+        is_datetime_name_str = isinstance(datetime_name, str)
+        is_datetime_name_int = isinstance(index_name, int)
+
+        if datetime_name in columns:
+            pos = columns.index(datetime_name)
+            columns[1], columns[pos] = columns[pos], columns[1]
+
+        elif datetime_name in data.columns:
+            index_name, *other_columns = columns
+            columns = [index_name, datetime_name, *other_columns]
+
+        elif (is_datetime_name_str or is_datetime_name_int) and datetime_name in data.index.names:
+            data = data.reset_index()
+            index_name, *other_columns = columns
+            columns = [index_name, datetime_name, *other_columns]
+
+        elif is_datetime_name_int and is_multiindex and datetime_name in unamed_index_levels:
+            data = data.reset_index()
+            datetime_name = f'level_{index_name}'
+            index_name, *other_columns = columns
+            columns = [index_name, datetime_name, *other_columns]
+
+        columns_info = []
+
+        for feature_name in t.cast(t.List[str], columns):
+            if feature_name == index_name:
+                kind = 'Index'
+            elif feature_name == label_name:
+                kind = t.cast(str, self.label_type).replace('_', ' ').capitalize()
+            elif feature_name in categorical_features:
+                kind = 'Categorical'
+            elif feature_name in numerical_features:
+                kind = 'Numerical'
+            elif feature_name == datetime_name:
+                kind = 'Datetime'
+            else:
+                kind = 'Unknown'
+
+            columns_info.append([
+                feature_name,
+                infer_dtype(data[feature_name], skipna=True),
+                kind
+            ])
+
+        columns_info = pd.DataFrame(
+            data=columns_info,
+            columns=['Column', 'DType', 'Kind'],
+        )
+
+        return columns, columns_info, data.loc[:, columns]
```

Comment:
why do you need `data.loc[:, columns]`? which columns will be filtered?

Reply:
1. I am doing `df.reset_index` above and it will add new columns
2. in order to show columns in desired order

in general, this method task is to prepare a dataframe in a form
```
| index | datetime? | *other columns | label? |
------------------------------------------------
```

Reply:
yes
`reset_index`, will produce a bunch of new columns, in case, if `multiindex` was used
and we need to get rid of them